### ISSUE TITLE
Incorrect message sizes are used when calculating transcript

### ISSUE THREAD

**Issue by steven-bellock** [DMTF/libspdm#2610]
From #2608.

In at least
https://github.com/DMTF/libspdm/blob/9bc69639f112be9d5bd8a4aa93cd0fe07d70361a/library/spdm_responder_lib/libspdm_rsp_certificate.c#L156 and https://github.com/DMTF/libspdm/blob/9bc69639f112be9d5bd8a4aa93cd0fe07d70361a/library/spdm_responder_lib/libspdm_rsp_digests.c#L166-L167 the transport's request size is used, which may be larger than the size of the parsed SPDM message. I looked through all the other messages and they appear to be OK, but it would be good if someone else could confirm that these are the only two instances of the issue.

**Comment by twilfredo** [comment]
> @alistair23 @twilfredo you haven't seen this on your end? Are you sending the exact SPDM sizes through the transport layer?

I _don't think_ we have run into this issue (@alistair23?),  at the transport layer level in `x__receive_message()/x_send_message()` we send the exact 'message_size'. Which is what I assume you mean by "exact SPDM size"?

**Comment by alistair23** [comment]
For reference you can see exactly what we are doing at: https://github.com/westerndigitalcorporation/spdm-utils

Wilfred shared where we convert `message_ptr` and `message_size` into a Rust slice We always send the entire buffer, so we send `message_size` bytes.

I'm not sure we would notice if we send more data though. From reading https://github.com/DMTF/libspdm/discussions/2608 it sounds like everything works, they are just noticing more data being sent. We aren't inspecting the bytes that closely, so as long as libspdm is happy we wouldn't notice

**Comment by aik** [comment]
> From reading #2608 it sounds like everything works, they are just noticing more data being sent. We aren't inspecting the bytes that closely, so as long as libspdm is happy we wouldn't notice

nope, it does not work, the CHALLENGE_AUTH response is not accepted by the requester, this how it's been found.

**Comment by alistair23** [comment]
Is there any chance you can share the log of your failure? I'm unable to reproduce it. The `CHALLENGE_AUTH` is accepted by a libspdm requester for me.

I understand this has now been fixed, but I would like to see if I can reproduce the failure (before the fix)
